client/src/hooks/useDownloadOptionsContext.js

```diff
@@ -46,7 +49,21 @@ export const useDownloadOptionsContext = () => {
 
   // Get the first computed file that matches modality and format
   const getFoundFile = (files = computedFiles) =>
-    files.find((file) => file.modality === modality && file.format === format)
+    files.find(
+      (file) =>
+        file.modality === modality &&
+        file.format === format &&
+        file.includes_merged === includesMerged
+    )
+
+  // Filter available computed files based on the selected modality
+  const filterComputedFilesByModality = (files = computedFiles) =>
+    files.filter((file) => file.modality === modality)
+
+  // Check the availability of the merged objects based on computedFilesByModality
+  const isMergedObjectsAvailable =
+    commputedFilesByModality &&
+    commputedFilesByModality.some((cf) => cf.includes_merged)
```

Comment:
```suggestion
  const isMergedObjectsAvailable = getFilteredFiles().some(
    (cf) => cf.includes_merged
  )
```

We actually want to take format into account. There could be merged `AnnData` but not merged `SingleCellExperiment` for any given `SingleCell` project. So we should always check if the modality and format match and then check if there is at least one where `includes_merged` is true.

Reply:
I updated the logic at 773b9f5485931310fbd3da08edd93a6ce3077ac4 (renamed the method at 6a05d4951cad4b56fc5fc87d2921d980f1dc9490).

In this update:
1. Filter the computed files based on the selected `modality,` `format`, and `includes_merged` to make sure that the merged objects is available for that specified combination above using the `getFilteredComputedFiles` method. 
2. Based on that filtered computed files, we set the checkbox availability and the warning text visibility.